packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -569,31 +569,143 @@ enum _StateLifecycle {
+///    means any calls to [setState] in [didUpdateConfig] are redundant.
+///  * If the subtree containing the [State] object is removed from the tree
+///    (e.g., because the parent built a widget with a different [runtimeType]
+///    or [key]), the framework calls the [deactivate] method. Subclasses
+///    should override this method to clean up any links between this object
+///    and other elements in the tree (e.g. if you have provided an ancestor
+///    with a pointer to a descendant's [RenderObject]).
+///  * At this point, the framework might reinsert this subtree into another
+///    part of the tree. If that happens, the framework will ensure that it
+///    calls [build] to give the [State] object a chance to adapt to its new
+///    location in the tree. If the framework does reinsert this subtree, it
+///    will do so before the end of the animation frame in which the subtree was
+///    removed from the tree. For this reason, [State] objects can differ
+///    releasing most resources until the framework calls their [dispose]
+///    method.
+///  * If the framework does not reinsert this subtree by the end of the current
+///    animation frame, the framework will call [dispose], which indiciates that
+///    this [State] object will never build again. Subclasses should override
+///    this method to release any resources retained by this object (e.g.,
+///    stop any active animations).
+///  * After the framework calls [dispose], the [State] object is considered
+///    unmounted and the [mounted] property is false. It is an error to call
+///    [setState] at this point. This stage of the lifecycle is terminal: there
+///    is no way to remount a [State] object that has been disposed.
+///
+/// See also:
+///
+///  * [StatefulWidget]
+///  * [StatelessWidget]
 @optionalTypeArgs
 abstract class State<T extends StatefulWidget> {
-  /// The current configuration (an instance of the corresponding
-  /// [StatefulWidget] class).
+  /// The current configuration.
+  ///
+  /// A [State] object's configuration is the corresponding [StatefulWidget]
+  /// instance. This property is initialized by the framework before calling
+  /// [initState]. If the parent updates this location in the tree to a new
+  /// widget with the same [runtimeType] and [key] as the current configuration,
+  /// the framework will update this property to refer to the new widget and
+  /// then call [didUpdateConfig], passing the old configuration as an argument.
   T get config => _config;
   T _config;
 
-  /// This is used to verify that State objects move through life in an orderly fashion.
+  /// The current stage in the lifecycle for this state object.
+  ///
+  /// This field is used by the framework when asserts are enabled to verify
+  /// that [State] objects move through their lifecycle in an orderly fashion.
   _StateLifecycle _debugLifecycleState = _StateLifecycle.created;
```

Comment:
it's obviously fine to doc private members, but did you intend to expose the value?